src/CMakeLists.txt

```diff
@@ -340,18 +332,16 @@ target_link_libraries(wesnoth-sdl
 
 ########### Wesnoth main source files ###############
 
-# Depending on whether the game or the game and the tests are build the main
-# sources are compiled in the game library or not. If only one target is needed
-# the putting in the game library solution is slower. If both targets are build
-# the library is faster (especially if the user doesn't use ccache (or
-# something similar)).
+# create libwesnoth-extras.a if building the client or the boost unit tests
 
-GetSources("wesnoth" wesnoth-main_SRC)
+if(ENABLE_GAME OR ENABLE_TESTS)
+	GetSources("libwesnoth_extras" wesnoth-main_SRC)
```

Comment:
I strongly dislike renaming source_lists/wesnoth to source_lists/libwesnoth_extras.

"Hmm, this file is only used by the client, I wonder where I should add it... Ah! libwesnoth_extras! That's *obviously* the right file!"

Reply:
Would "wesnoth_client" be acceptable?

Reply:
Yes, but I'd prefer just "wesnoth" like now.